guides/hack/03-expressions-and-operators/01-introduction.md

```diff
@@ -1,29 +1,43 @@
-* [list intrinsic function](list.md)
-* [Logical operators, `&&`, `||`, `!`](logical-operators.md)
-* [Member selection, `->`, `?->`](member-selection.md)
-* [new](new.md)
-* [Operator precedence](operator-precedence.md)
-* [Pipe, `|>`](pipe.md)
-* [Scope resolution, `::`](scope-resolution.md)
-* [Spaceship, `<=>`](equality.md)
-* [String concatenation, `.`](string-concatenation.md)
-* [Subscripting, `[]`](subscript.md)
-* [Ternary, `?` `:`](ternary.md)
-* [Type Assertions `is`, `as`, `?as`](type-assertions.md)
-* [Yield](yield.md)
+## Assignment Operators
+* [Assignment](/hack/expressions-and-operators/assignment) (`=`, `+=`, and more)
+* [Bitwise](/hack/expressions-and-operators/bitwise-operators) (`&`, `|`, `^`, `<<`, `>>`, `~`)
+* [Coalesce](/hack/expressions-and-operators/coalesce) (`??`, `??=`)
+* [Pipe](/hack/expressions-and-operators/pipe) (`|>` then stored in `$$`)
+* [Spaceship](/hack/expressions-and-operators/equality#the-spaceship-operator) (`<=>`)
+* [Ternary](/hack/expressions-and-operators/ternary) (`?` `:`, and `?:`)
+
+## Comparison Operators
+* [Comparison](/hack/expressions-and-operators/comparisons) (`>`, `>=`,`<`,`<=`)
+* [Equality Comparison](/hack/expressions-and-operators/equality) (`==`, `!=`, `===`, and more)
+* [Logical Comparison](/hack/expressions-and-operators/logical-operators) (`&&`, `||`, `!`)
+
+## Math Operators
+* [Arithmetic](/hack/expressions-and-operators/arithmetic) (`+`, `-`, `*`, `/`,  and more)
+* [Increment](/hack/expressions-and-operators/incrementing-and-decrementing) (`++`)
+* [Decrement](/hack/expressions-and-operators/incrementing-and-decrementing) (`--`)
+
+## Class Operators
+* [Create an Object](/hack/expressions-and-operators/new) (`new`)
+* [Access Instance Members](/hack/expressions-and-operators/member-selection) (`->`, `?->`)
+* [Access Static Members / Enum Properties](/hack/expressions-and-operators/scope-resolution) (`::`)
+* [Access XHP Attributes](/hack/expressions-and-operators/XHP-attribute-selection) (`->:`)
+
+## Type Operators
+* [Cast/Convert Types](/hack/expressions-and-operators/casting) (`(int)`, `(string)`, and more)
+* [Check Types](/hack/expressions-and-operators/type-assertions#checking-types-with-is) (`is`)
+* [Enforce Types](/hack/expressions-and-operators/type-assertions#enforcing-types-with-as-and-as) (`as`, `?as`)
+* [Concatenate a String](/hack/expressions-and-operators/string-concatenation) (`.`)
+* [Index Hack Arrays and Strings](/hack/expressions-and-operators/subscript) (`[]`)
+* [Unpack Tuples](/hack/expressions-and-operators/list) (`list`)
+
+## Function Operators
+* [Async: Suspend Execution](/hack/expressions-and-operators/await) (`await`)
+* [Define a Generator](/hack/expressions-and-operators/yield) (`yield`)
+* [Write to Standard Output](/hack/expressions-and-operators/echo) (`echo`)
+* [Terminate](/hack/expressions-and-operators/exit) (`exit`)
+
+## Error Operators
+* [Indicate Errors](/hack/expressions-and-operators/invariant) (`invariant`)
```

Comment:
invariant() is an unconditional assert().
Suggestion for the link text "check an invariant / assert and invariant"